src/frontend/utils/conversejs/converse-plugins/chatPlugin.ts

```diff
@@ -9,6 +27,62 @@ const addChatPlugin = (xmpp: XMPP) =>
     initialize() {
       const _converse = this._converse;
 
+      _converse.on('initialized', () => {
+        _converse.connection.addHandler(
+          (stanza: HTMLElement) => {
+            switch (stanza.nodeName) {
+              case StanzaType.Message:
+                switch (determineMessageType(stanza)) {
+                  case StanzaMessageType.ClassiqueMessage:
+                    const msg: ChatMessageType = {
+                      dateTime: DateTime.now(),
+                      sender: getNameFromJID(stanza.getAttribute('from')!),
+                      content:
+                        stanza.getElementsByTagName('body')[0].textContent!,
+                    };
+                    useMessagesState.getState().addMessage(msg);
+                    break;
+                  case StanzaMessageType.History:
+                    const oldMsgStanza =
+                      stanza.getElementsByTagName('message')[0];
+                    const oldMsg: ChatMessageType = {
+                      dateTime: getMsgDateTimeFromStanza(stanza),
+                      sender: getNameFromJID(
+                        oldMsgStanza.getAttribute('from')!,
+                      ),
+                      content:
+                        oldMsgStanza.getElementsByTagName('body')[0]
+                          .textContent!,
+                    };
+                    useMessagesState.getState().addMessage(oldMsg);
+                    break;
+                }
+
+                break;
+
+              case StanzaType.Presence:
+                break;
+
+              case StanzaType.Iq:
+                break;
+
+              default:
+                // tslint:disable-next-line: no-console
+                console.error('Unrecognized stanza');
```

Comment:
You should use the `report` function here instead of `console.error` IMO

Reply:
fix in [19b586f91898044a5a045a35d967691918afc30e](https://github.com/openfun/marsha/pull/1247/commits/19b586f91898044a5a045a35d967691918afc30e)